penrose-web/src/engine/Evaluator.ts

```diff
@@ -206,11 +196,13 @@ export const evalExprs = (
   optDebugInfo?: OptDebugInfo
 ): ArgVal<VarAD>[] => es.map((e) => evalExpr(e, trans, varyingVars, optDebugInfo));
 
-function toFloatVal<T>(a: ArgVal<T>): T {
+function toFloatVal<VarAD>(a: ArgVal<VarAD>): VarAD {
   if (a.tag === "Val") {
     const res = a.contents;
     if (res.tag === "FloatV") {
       return res.contents;
+    } else if (res.tag === "IntV") {
+      return (constOf(res.contents) as unknown) as VarAD; // Not sure why TS compiler has problem here
```

Comment:
Umm it's not recognizing `res.contents` has to be `number` here?

Reply:
Oh, I fixed it. L199 had a fake `<VarAD>` type parameter here which was causing a super weird error message. Removing that fixes it.